enterprise/coderd/userauth.go

```diff
@@ -0,0 +1,50 @@
+package coderd
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/codersdk"
+)
+
+func (api *API) setUserGroups(ctx context.Context, tx database.Store, userID uuid.UUID, groupNames []string) error {
```

Comment:
I think we rename `tx` to `db` and then use `db.InTx` so that we guarantee this runs in a transaction regardless of whether the calling function has opened one or not